Library/Homebrew/formulary.rb

```diff
@@ -97,17 +97,31 @@ def load_file
   class BottleLoader < FormulaLoader
     def initialize bottle_name
       @bottle_filename = Pathname(bottle_name).realpath
-      name_without_version = bottle_filename_formula_name @bottle_filename
-      if name_without_version.empty?
-        if ARGV.homebrew_developer?
-          opoo "Add a new regex to bottle_version.rb to parse this filename."
+      @bottle_version = BottleVersion.parse @bottle_filename
+      name = full_name = nil
+      success = catch(:done) do
+        if @bottle_version.name.empty? || @bottle_version.version.nil?
+          name = full_name = bottle_name
+          throw(:done, false)
         end
-        name = bottle_name
-      else
-        name = name_without_version
+
+        name = full_name = @bottle_version.name
+        version = @bottle_version.version
+        tabfile = Utils.popen_read("tar", "-xOzf", @bottle_filename, "#{name}/#{version}/INSTALL_RECEIPT.json")
```

Comment:
It's doable. We can use below code to get version and formula name.
```ruby
receipt_file = Utils.popen_read("tar", "-tzf", bottle_file, "*/*/INSTALL_RECEIPT.json").chomp
name, version, _ = receipt_file.split("/", 3)
```

However, it requires large code clean so I plan to create a new PR. Also, I will change the format of `*.bottle.rb` to include formula name if we want to remove `BottleVersion` completely